src/core/server/status/log_overall_status.ts

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { Observable } from 'rxjs';
+import { distinctUntilChanged, pairwise, startWith, takeUntil, map } from 'rxjs/operators';
+import { ServiceStatus } from './types';
+
+export const getOverallStatusChanges = (
+  overall$: Observable<ServiceStatus>,
+  stop$: Observable<void>
+) => {
+  return overall$.pipe(
+    takeUntil(stop$),
+    distinctUntilChanged((previous, next) => {
+      return previous.level.toString() === next.level.toString();
+    }),
+    startWith(undefined),
+    pairwise(),
+    map(([oldStatus, newStatus]) => {
+      if (oldStatus) {
+        return `Kibana is now ${newStatus!.level.toString()} (was ${oldStatus!.level.toString()})`;
+      }
+      return `Kibana is now ${newStatus!.level.toString()}`;
```

Comment:
NIT: Should we follow a similar convention to Elasticsearch? The `[GREEN]`/`[YELLOW]`/`[RED]` approach?

Reply:
It has a bit different meaning in Elasticsearch. https://www.elastic.co/guide/en/elasticsearch/reference/current/cluster-health.html
```
On the shard level, a red status indicates that the specific shard is not allocated in the cluster, yellow means that the primary shard is allocated but replicas are not, and green means that all shards are allocated. 
```
IMO it's rather confusing in the case of Kibana since we don't operate any shards nor replicas.

Reply:
That was the old/legacy system, and we moved away from it in KP, so I guess not. @joshdover may know better though.